backend/djangoindia/api/views/event.py

```diff
@@ -3,19 +3,35 @@
     EventSerializer,
 )
 from djangoindia.bg_tasks.event_registration import registration_confirmation_email_task
-from djangoindia.db.models.event import Event, EventRegistration
+from djangoindia.db.models.event import Event, EventRegistration,Sponsorship
 from rest_framework import generics, status
 from rest_framework.mixins import CreateModelMixin, ListModelMixin, RetrieveModelMixin
 from rest_framework.response import Response
 
 from djangoindia.constants import POST, PRIMARY_KEY_SHORT
+from django.db.models import Prefetch
 
 
 # Create your views here.
 class EventAPIView(
     generics.GenericAPIView, ListModelMixin, CreateModelMixin, RetrieveModelMixin
 ):
-    queryset = Event.objects.all()
+    queryset = Event.objects.all().prefetch_related(
+        Prefetch(
+            'sponsors',
+            queryset=Sponsorship.objects.filter(
+                sponsorship_type='event-sponsorship'
+            ).select_related('sponsor').only(
+                'sponsorship_tier',
+                'sponsor__url',
+                'sponsor__name',
+                'sponsor__type',
+                'sponsor__contact_email',
+                'sponsor__logo'

```

Comment:
why two underscores?

Reply:
Its a way to access attributes of related model in django.

Reply:
gotcha. thanks.